src/sentry/search/snuba/executors.py

```diff
@@ -71,134 +40,184 @@ def get_search_filter(search_filters, name, operator):
     return found_val
 
 
+@six.add_metaclass(ABCMeta)
 class AbstractQueryExecutor:
     """This class serves as a template for Query Executors.
     We subclass it in order to implement query methods (we use it to implement two classes: joined Postgres+Snuba queries, and Snuba only queries)
     It's used to keep the query logic out of the actual search backend,
     which can now just build query parameters and use the appropriate query executor to run the query
     """
 
-    def query(
-        self,
-        projects,
-        retention_window_start,
-        group_queryset,
-        environments,
-        sort_by,
-        limit,
-        cursor,
-        count_hits,
-        paginator_options,
-        search_filters,
-        date_from,
-        date_to,
-    ):
-        raise NotImplementedError
+    EMPTY_RESULT = Paginator(Group.objects.none()).get_result()
```

Comment:
Just to be on the safe side, I would not make this a class attribute, since, when you return it, you will return a reference to this object, you will not be building a new one. If the Paginator itself is mutable, someone receiving it could make changes that will impact everyone else even on a different instance of AbstractQueryExecutor

Reply:
Changed to
`@property
  def empty_result(self):
      return Paginator(Group.objects.none()).get_result()`